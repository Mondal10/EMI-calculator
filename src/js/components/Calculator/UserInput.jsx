import React from 'react';

import styled from 'styled-components';

const InputContainer = styled.div`
  display: flex;
  justify-content: space-between;
  font-size: 20px;
  padding: 2% 0;
  color: #082f52;
`;

const Lable = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Input = styled.input.attrs({
  type: 'text',
})`
  color: #082f52;
  font-size: 18px;
  cursor: pointer;
  border-radius: 5px;
  height: 35px;
  border: 2px solid #A3BDE6;
`;

function UserInput({ name, label, value, handleInput }) {

  const handleOnkeyDown = (event) => {
    const { keyCode } = event;

    // All keys except numbers, arrow navigation, delete, backspace and tab are blocked
    if (!((keyCode >= 48 && keyCode <= 57) || (keyCode >= 37 && keyCode <= 40) || keyCode == 8 || keyCode == 46 || keyCode == 9 || keyCode == 190 || keyCode == 110)) {
      event.preventDefault();
    }
  };

  return (
    <InputContainer>
      <Lable>{label}</Lable>
      <div>
        <Input name={name} value={value} onChange={handleInput} onKeyDown={handleOnkeyDown} />
      </div>
    </InputContainer>
  )
}

export default UserInput;
