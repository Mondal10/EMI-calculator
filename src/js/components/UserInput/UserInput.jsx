import React from 'react';

import styled from 'styled-components';

const InputContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  font-size: 20px;
  padding: 2% 0;
  color: #e8e9e9;
`;

const Lable = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Input = styled.input.attrs({
  type: 'text',
})`
  color: #082f52;
  font-size: 18px;
  cursor: pointer;
  border-radius: 5px;
  height: 35px;
  border: 2px solid #A3BDE6;
`;

const InfoSpan = styled.span`
  padding: 7px;
`;

function UserInput({ name, symbol, label, value, handleInput }) {

  return (
    <InputContainer>
      <Lable>{label}</Lable>
      <div>
        <label>
          <InfoSpan>{symbol}</InfoSpan>
          <Input name={name} value={value} onChange={handleInput} />
        </label>
      </div>
    </InputContainer>
  )
}

export default UserInput;
