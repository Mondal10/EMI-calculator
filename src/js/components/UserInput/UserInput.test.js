import React from 'react';
import renderer from 'react-test-renderer';
import UserInput from './UserInput';

describe("UserInput Snapshot", () => {
  it('Loan Amount UserInput rendered successfully', () => {
    const tree = renderer.create(<UserInput name='LA' symbol='â‚¹' value='1000000' label='Loan Amount' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Loan Tenure UserInput rendered successfully', () => {
    const tree = renderer.create(<UserInput name='LT' symbol="Mo" value='240' label='Loan Tenure' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
