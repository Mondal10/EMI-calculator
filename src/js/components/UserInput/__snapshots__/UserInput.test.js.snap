// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInput Snapshot Loan Amount UserInput rendered successfully 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-size: 20px;
  padding: 2% 0;
  color: #e8e9e9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  color: #082f52;
  font-size: 18px;
  cursor: pointer;
  border-radius: 5px;
  height: 35px;
  border: 2px solid #A3BDE6;
}

.c2 {
  padding: 7px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Loan Amount
  </div>
  <div>
    <label>
      <span
        className="c2"
      >
        â‚¹
      </span>
      <input
        className="c3"
        name="LA"
        type="text"
        value="1000000"
      />
    </label>
  </div>
</div>
`;

exports[`UserInput Snapshot Loan Tenure UserInput rendered successfully 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-size: 20px;
  padding: 2% 0;
  color: #e8e9e9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  color: #082f52;
  font-size: 18px;
  cursor: pointer;
  border-radius: 5px;
  height: 35px;
  border: 2px solid #A3BDE6;
}

.c2 {
  padding: 7px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Loan Tenure
  </div>
  <div>
    <label>
      <span
        className="c2"
      >
        Mo
      </span>
      <input
        className="c3"
        name="LT"
        type="text"
        value="240"
      />
    </label>
  </div>
</div>
`;
